File: SQL/Views/SQLCreateView.vb
' ___________________________________________________
'
'  © Hi-Integrity Systems 2012. All rights reserved.
'  www.hisystems.com.au - Toby Wicks
' ___________________________________________________
'

Option Strict On
Option Explicit On

Namespace SQL

    Public Class SQLCreateView
        Inherits SQLStatement

        Private pstrName As String
        Private pobjSelect As SQLSelect

        Public Sub New()

        End Sub

        Public Sub New(ByVal strViewName As String, ByVal objSelectStatement As SQLSelect)

            Me.Name = strViewName
            Me.Select = objSelectStatement

        End Sub

        Public Property Name() As String
            Get

                Return pstrName

            End Get

            Set(ByVal Value As String)

                If String.IsNullOrEmpty(Value) Then
                    Throw New ArgumentNullException()
                End If

                pstrName = Value

            End Set
        End Property

        Public Property [Select] As SQLSelect
            Get

                Return pobjSelect

            End Get

            Set(value As SQLSelect)

                If value Is Nothing Then
                    Throw New ArgumentNullException
                End If

                pobjSelect = value

            End Set
        End Property

        Public Overrides ReadOnly Property SQL() As String
            Get

                If String.IsNullOrEmpty(Me.Name) Then
                    Throw New Exceptions.DatabaseObjectsException("View name has not been set")
                ElseIf pobjSelect Is Nothing Then
                    Throw New Exceptions.DatabaseObjectsException("Select statement has not been set")
                End If

                Return _
                    "CREATE VIEW " & _
                    SQLConvertIdentifierName(Me.Name, Me.ConnectionType) & " AS " & _
                    pobjSelect.SQL

            End Get
        End Property

    End Class

End Namespace

File: SQL/Views/SQLCreateView.cs
// _________________________________________________________________________
//
//  © Hi-Integrity Systems 2012. All rights reserved.
//  www.hisystems.com.au - Toby Wicks
// 
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//	    http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
// _________________________________________________________________________
//

using System.Collections;
using System;
using System.Data;

namespace DatabaseObjects.SQL
{
	public class SQLCreateView : SQLStatement
	{
		private string pstrName;
		private SQLSelect pobjSelect;
			
		public SQLCreateView()
		{
		}
			
		public SQLCreateView(string strViewName, SQLSelect objSelectStatement)
		{
			this.Name = strViewName;
			this.Select = objSelectStatement;
		}
			
		public string Name
		{
			get
			{
				return pstrName;
			}
				
			set
			{
				if (string.IsNullOrEmpty(value))
					throw new ArgumentNullException();
					
				pstrName = value;
			}
		}
			
		public SQLSelect Select
		{
			get
			{
				return pobjSelect;
			}
				
			set
			{
				if (value == null)
					throw new ArgumentNullException();
					
				pobjSelect = value;
			}
		}
			
		public override string SQL
		{
			get
			{
				return base.Serializer.SerializeCreateView(this);
			}
		}
	}
}
	
